c/tskit/tables.c

```diff
@@ -12346,117 +12502,32 @@ tsk_table_collection_deduplicate_sites(
 
 int TSK_WARN_UNUSED
 tsk_table_collection_compute_mutation_parents(
-    tsk_table_collection_t *self, tsk_flags_t TSK_UNUSED(options))
+    tsk_table_collection_t *self, tsk_flags_t options)
 {
     int ret = 0;
-    tsk_id_t num_trees;
-    const tsk_id_t *I, *O;
-    const tsk_edge_table_t edges = self->edges;
-    const tsk_node_table_t nodes = self->nodes;
-    const tsk_site_table_t sites = self->sites;
     const tsk_mutation_table_t mutations = self->mutations;
-    const tsk_id_t M = (tsk_id_t) edges.num_rows;
-    tsk_id_t tj, tk;
-    tsk_id_t *parent = NULL;
-    tsk_id_t *bottom_mutation = NULL;
-    tsk_id_t u;
-    double left, right;
-    tsk_id_t site;
-    /* Using unsigned values here avoids potentially undefined behaviour */
-    tsk_size_t j, mutation, first_mutation;
-
-    /* Set the mutation parent to TSK_NULL so that we don't check the
-     * parent values we are about to write over. */
-    tsk_memset(mutations.parent, 0xff, mutations.num_rows * sizeof(*mutations.parent));
-    num_trees = tsk_table_collection_check_integrity(self, TSK_CHECK_TREES);
-    if (num_trees < 0) {
-        ret = (int) num_trees;
-        goto out;
-    }
-    parent = tsk_malloc(nodes.num_rows * sizeof(*parent));
-    bottom_mutation = tsk_malloc(nodes.num_rows * sizeof(*bottom_mutation));
-    if (parent == NULL || bottom_mutation == NULL) {
-        ret = tsk_trace_error(TSK_ERR_NO_MEMORY);
-        goto out;
-    }
-    tsk_memset(parent, 0xff, nodes.num_rows * sizeof(*parent));
-    tsk_memset(bottom_mutation, 0xff, nodes.num_rows * sizeof(*bottom_mutation));
-    tsk_memset(mutations.parent, 0xff, self->mutations.num_rows * sizeof(tsk_id_t));
 
-    I = self->indexes.edge_insertion_order;
-    O = self->indexes.edge_removal_order;
-    tj = 0;
-    tk = 0;
-    site = 0;
-    mutation = 0;
-    left = 0;
-    while (tj < M || left < self->sequence_length) {
-        while (tk < M && edges.right[O[tk]] == left) {
-            parent[edges.child[O[tk]]] = TSK_NULL;
-            tk++;
-        }
-        while (tj < M && edges.left[I[tj]] == left) {
-            parent[edges.child[I[tj]]] = edges.parent[I[tj]];
-            tj++;
-        }
-        right = self->sequence_length;
-        if (tj < M) {
-            right = TSK_MIN(right, edges.left[I[tj]]);
-        }
-        if (tk < M) {
-            right = TSK_MIN(right, edges.right[O[tk]]);
+    if (!(options & TSK_NO_CHECK_INTEGRITY)) {
+        /* Set the mutation parent to TSK_NULL so that we don't error
+           in the integrity check on data we're about to overwite.
+        */
+        tsk_memset(
```

Comment:
I don't think we need to do this, and we should't update the data if we error. TSK_CHECK_TREES doesn't imply TSK_CHECK_MUTATION_PARENTS

Reply:
Good point about not changing if we error. Here I was thinking about the fact that `TSK_CHECK_TREES` implies `TSK_CHECK_MUTATION_ORDERING` which checks order based on the existing parent column.

Fixed in 829c9ce6